src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IArgument.vb

```diff
@@ -746,6 +746,42 @@ BC30057: Too many arguments to 'Public Sub M2(x As Integer, [y As Integer = 0],
 
         <CompilerTrait(CompilerFeature.IOperation)>
         <Fact()>
+        Public Sub TestValidDynamicInvocation_OmittedArgument()
```

Comment:
>TestValidDynamicInvocation_OmittedArgument [](start = 19, length = 42)

It would be good to have a test for a scenario when an omitted argument is used in a successful method call. In particular we should verify what IOperation nodes are going to be associated with OmittedArgumentSyntax.

Reply:
I mean an early bound scenario.

Reply:
> I mean an early bound scenario.

@AlekseyTs, do you have an example? I wasn't able to come up with such a valid/compiling code where OmittedArgumentSyntax is used.

Reply:
It looks like we already have a test for this scenario - DirectlyBindOmittedArgument_InvocationExpression. I think the following nodes
```
          IArgument (ArgumentKind.DefaultValue, Matching Parameter: b) (OperationKind.Argument) (Syntax: 'M2')
            ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 0) (Syntax: 'M2')
```
should be associated with OmittedArgumentSyntax, which isn't the case right now.

Reply:
Ah, I noticed these were getting exposed as IArgument with ArgumentKind.DefaultValue but did not notice the syntax was getting mapped to method name instead of OmittedArgumentSyntax. I will fix the issue and update the tests.

Reply:
I debugged into this and it seems we need to change `OverloadResolution.MatchArguments` over [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic/Semantics/OverloadResolution.vb,3073) to pass in the `OmittedArgumentSyntax` instead of `methodOrPropertyGroup.Syntax`, which is not trivial as `argIndex = -1` and `argument = Nothing` at this point. I will open a separate issue to track this bug.

Reply:
Filed #22486